src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -158,6 +159,80 @@ internal static void DisposeAll(this IEnumerable<IDisposable> disposables)
                 disposable.Dispose();
             }
         }
+
+        internal static bool IsSelfSigned(this X509Certificate2 cert)
+        {
+            X509SubjectKeyIdentifierExtension skid = null;
+            X509AuthorityKeyIdentifierExtension akid = null;
+
+            foreach (X509Extension extension in cert.Extensions)
+            {
+                if (extension.Oid.Value == Oids.SubjectKeyIdentifier)
+                {
+                    skid = (X509SubjectKeyIdentifierExtension)extension;
+                }
+                else if (extension.Oid.Value == Oids.AuthorityKeyIdentifier)
+                {
+                    akid = new X509AuthorityKeyIdentifierExtension();
+                    akid.CopyFrom(extension);
+                }
+
+                if (akid != null && skid != null)
+                {
+                    break;
+                }
+            }
+
+            return IsSelfSigned(cert, skid, akid);
+        }
+
+        internal static bool IsSelfSigned(
+            this X509Certificate2 cert,
+            X509SubjectKeyIdentifierExtension skid,
+            X509AuthorityKeyIdentifierExtension akid)
+        {
+            Debug.Assert(cert != null);
+
+            if (!cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData))
```

Comment:
Do the subjects actually have to be byte-for-byte equal? For example, could I switch the parts of a DN and have an equivalent name?